web/client/reducers/maps.js

```diff
@@ -243,23 +244,10 @@ function maps(state = {
         let newState = assign({}, state, {
             results: newMaps.map(function(map) {
                 if (map.id === action.mapId) {
-
-                    // Fix to overcome GeoStore bad encoding of single object arrays
-                    let fixedSecurityRule = [];
-                    if (action.permissions && action.permissions.SecurityRuleList && action.permissions.SecurityRuleList.SecurityRule) {
-                        if ( isArray(action.permissions.SecurityRuleList.SecurityRule)) {
-                            fixedSecurityRule = action.permissions.SecurityRuleList.SecurityRule;
-                        } else {
-                            fixedSecurityRule.push(action.permissions.SecurityRuleList.SecurityRule);
-                        }
-                    }
-
                     return assign({}, map, {
                         permissionLoading: false,
                         permissions: {
-                            SecurityRuleList: {
-                                SecurityRule: fixedSecurityRule
-                            }
+                            ...fixPermissions(action.permissions)
```

Comment:
why this fix permission ?

Reply:
I just refactored the logic that was already there in a function to reuse.